[RR] Add Interactive settings and Minor Release Rebase

PlaidCat opened the pull request:
The new feature allows a user to run in interactive mode where the rolling release will "pause" while the engineer fixes the failed forward-port and commit.  It will then attempt to continue.

Actual Run https://github.com/ctrliq/kernel-src-tree/pull/684

PlaidCat committed:
[KRU] Allow rolling release to be used across centos

Previously the tooling would find the common RESF tag but when upgrading
minor versions it would not search across centos versions.

PlaidCat committed:
[RR] Add Interactive settings

The new feature allows a user to run in interactive mode where the
rolling release will "pause" while the engineer fixes the failed
forward-port and commit.  It will then attempt to continue.

bmastbergen commented on rolling-release-update.py:
```diff
@@ -301,7 +303,73 @@ def check_for_fips_protected_changes(repo, branch, common_tag):
+                        print('[rolling release update] Please complete the cherry-pick with:')
+                        print('[rolling release update]   git cherry-pick --continue')
+                        print('[rolling release update] or abort with:')
+                        print('[rolling release update]   git cherry-pick --abort')
```
or skip with: git cherry-pick --skip ?

PlaidCat replied:
Will add the RUFF changes caused a pretty big conflict that I'm resolving I'll also be including the `new minor version` update as well when I come back to this

roxanan1996 replied:
Yeah, we should have merged all of these before that. Sorry.

PlaidCat replied:
No worries this happens when we have so many plates in the air

bmastbergen reviewed (commented):
LGTM, but maybe add the --skip suggestion (you used that multiple times in the demo yesterday)

roxanan1996 reviewed (dismissed):
LGTM

PlaidCat committed:
[RR] Fix CentOS rebase check

One of the critical items as a fork of RedHat is that our kernel updates
out from under us and some of the features/fixes/etc we backport in one
version of RLC might need to be dropped on the next update.  The latest
minor update, to 10.1, rebase missed some items that should have been
dropped due being present in in CentOS streams branch but not a part of
the RHEL minor version updates.

WIWT: Fixed some debugging.

PlaidCat committed:
[RR] RUFF formating updates

PlaidCat commented on rolling-release-update.py:
```diff
@@ -342,6 +408,80 @@ def check_for_fips_protected_changes(repo, branch, common_tag):
+
+                        # Print remaining commits including the current failed one
+                        remaining_commits = list(reversed(rolling_commit_map.items()))
+                        start_idx = remaining_commits.index((ciq_commit, upstream_commit))
```
I"m not sure how or why the order would change so i'm not sure if this is possible ... its true since this is aborting maybe its worth doing. 🤔

PlaidCat committed:
[RR] Fix up potential index exception
